tests/rainbowGrades/test_csci1100.py

```diff
@@ -0,0 +1,295 @@
+
+    # Use the same method a user would to pull reports from Submitty to Rainbow Grades
+    print("Attempting to rsync contents")
+    os.chdir(summary_tmp)
+    return_code = subprocess.call(["make", "pull"])
+
+    if return_code != 0:
+        error_and_cleanup(test_tmp, "Failed to rsync data (Error {})",format(return_code))
+
+    if not os.path.exists(os.path.join(summary_tmp, "raw_data")):
+        error_and_cleanup(test_tmp, "Could not find raw_data folder after rsync'ing")
+
+    # Extract the test version of the reports for comparison
+    print("Extracting known raw data")
+    known_raw_path = os.path.join(test_tmp, "raw_data")
+    summary_raw_path = os.path.join(summary_tmp, "raw_data")
+    os.mkdir(known_raw_path)
+    return_code = subprocess.call(["tar", "-xf", os.path.join(script_path, "raw_data_10090542_csci1100.tar"),
+                                   "-C", known_raw_path])
+    if return_code != 0:
+        error_and_cleanup(test_tmp, "Extracting raw data failed (Error {}".format(return_code))
+    print("Comparing known raw data to rsync'd raw data")
+
+    # Construct a list of all files in the Submitty and test versions of reports to make sure the name/number matches
+    known_files = []
+    summary_files = []
+    for f in os.listdir(known_raw_path):
+        known_files.append(f)
+    for f in os.listdir(summary_raw_path):
+        summary_files.append(f)
+
+    if len(known_files) != len(summary_files):
+        file_diff = len(known_files) - len(summary_files)
+        if len(known_files) > len(summary_files):
+            error_and_cleanup(test_tmp,
+                              "There are {} more files in the rsync'd raw_data than expected.".format(file_diff))
+        else:
+            error_and_cleanup(test_tmp,
+                              "There are {} fewer files in the rsync'd raw_data than expected.".format(-1 * file_diff))
+
+    # Verify the content (except for time-dependent fields) of Submitty raw_data files match with test version
+    for f in known_files:
+        contents1 = ""
+        contents2 = ""
+        filename1 = os.path.join(known_raw_path, f)
+        filename2 = os.path.join(summary_raw_path, f)
+
+        try:
+            with open(filename1, 'r') as file1:
+                contents1 = file1.readlines()
+
+            with open(filename2, 'r') as file2:
+                contents2 = file2.readlines()
+        except Exception as e:
+            error_and_cleanup(test_tmp, "{}".format(e))
+
+        # Use filters to avoid time-dependent fields and speed up comparison
+        filter1 = filter(remove_extra_raw_data_fields, contents1)
+        filter2 = filter(remove_extra_raw_data_fields, contents2)
+        same_flag = True
+        for x, y in zip(filter1, filter2):
+            if x != y:
+                same_flag = False
+        if not same_flag:
+            error_and_cleanup(test_tmp, "{} and {} differ".format(filename1, filename2))
```

Comment:
do you really need to go through the whole for loop before issuing this function? Can we not call this the first time that `x != y`?